x-pack/platform/plugins/shared/cases/server/routes/api/utils.ts

```diff
@@ -35,7 +41,45 @@ export function wrapError(
   };
 }
 
-export const escapeHatch = schema.object({}, { unknowns: 'allow' });
+export const escapeHatch = schema.object(
+  {},
+  {
+    unknowns: 'allow',
+  }
+);
+
+export const validAttachment = schema.object(
+  {},
+  {
+    unknowns: 'allow',
+    validate: (value) => {
+      if (
+        isPersistableStatePageAttachment(value) &&
+        value.persistableStateAttachmentTypeId === PAGE_ATTACHMENT_TYPE
+      ) {
+        const persistedState =
+          value.persistableStateAttachmentState as PageAttachmentPersistedState;
+        const url = persistedState?.url?.pathAndQuery;
+        if (url && !isInternalURL(url)) {
+          return `External urls are not supported for page attachments. The provided url is: ${url}`;
+        }
+      }
+    },
+  }
+);
+
+/**
+ * Checks if the given value is a persistable state page attachment type.
+ */
+export const isPersistableStatePageAttachment = (
```

Comment:
nit: This isn't actually checking if it's a `pageAttachment` type, just that it's a persistableState type and is a valid value. Not sure if you want to move `value.persistableStateAttachmentTypeId === PAGE_ATTACHMENT_TYPE` here

Reply:
This particularly util wasn't intended to check if it's a `pageAttachment`. Just that it's a persistable state attachment type so that the types would later recognize `value.persistableStateAttachmentTypeId`. 

I suppose I could move all the checks here. That just wasn't my original intent.